web/src/components/summary/PairStatHistogram.vue

```diff
@@ -142,99 +146,112 @@ const draw = (): void => {
   // Clear the histogram.
   histogramContent.selectAll("*").remove();
 
-  // Create the axes.
-  const xScale = d3
+  // X-axis
+  const x = d3
     .scaleLinear()
-    .domain([0, 1] as [number, number])
+    .domain([0, 1])
     .range([0, width.value]);
-  const domain = xScale.domain();
-  const ticks = xScale.ticks(props.ticks);
-  const adjustedTicks = ticks[ticks.length - 1] === domain[1] ? ticks.slice(0, -1) : ticks;
+  const xAxis = histogramContent
+    .append("g")
+    .attr("transform", "translate(0," + height.value + ")")
+    .call(d3.axisBottom(x).tickFormat(d3.format(".0%")))
+    .classed("d3-ticks", true);
+  xAxis
+    .append("text")
+    .text(fieldName.value)
+    .classed("d3-label", true)
+    .attr("transform", `translate(${width.value / 2}, 50)`);
+
+  // X-ticks
+  const xDomain = x.domain();
+  const xTicks = x.ticks(props.ticks);
+  const xTicksAjusted = xTicks[xTicks.length - 1] === xDomain[1] ? xTicks.slice(0, -1) : xTicks;
+
+  // Histogram
   const histogram = d3
     .bin()
     .domain([0, 1])
-    .thresholds(adjustedTicks);
+    .thresholds(xTicksAjusted);
+  // Bins
   const bins = histogram(maxFileData.value);
-  const yScale = d3
+  
+  // Y-axis
+  const y = d3
     .scaleLinear()
     .range([height.value, 0])
-    .domain([
-      0,
-      d3.max<d3.Bin<number, number>, number>(bins, ({ length }) => length),
-    ] as number[]);
-  histogramContent
-    .append("g")
-    .attr("transform", "translate(0," + height.value + ")")
-    .call(d3.axisBottom(xScale));
-  histogramContent
+    .domain([0, d3.max<any, any>(bins, d => d.length)]);
+  const yAxis = histogramContent
     .append("g")
-    .call(d3.axisLeft(yScale));
-
-  histogramXScale.value = xScale;
-  histogramYScale.value = yScale;
+    .call(d3.axisLeft(y))
+    .classed("d3-ticks", true);
+  yAxis
+    .append("text")
+    .text("Amount of files")
+    .classed("d3-label", true)
+    .attr("transform", `translate(-55, ${height.value / 2 + 35}) rotate(90)`);
 
   // Add the data.
-  histogramContent
-    .selectAll("rect")
-    .data<d3.Bin<number, number>>(bins)
-    .enter()
-    .append("rect")
-    .attr("x", 1)
-    .attr("transform", (d) => {
-      return "translate(" + xScale(d.x0 || 0) + "," + yScale(d.length) + ")";
-    })
-    .attr("width", (d) => {
-      return Math.max(xScale(d.x1 || 0) - xScale(d.x0 || 0) - 1, 0);
-    })
-    .attr("height", (d) => {
-      return height.value - yScale(d.length);
-    })
-    .style("fill", d => getBinColor(d));
+  // Only add the data if there are any files available.
+  if (maxFileData.value.length > 0) {
```

Comment:
This chart is very similar to the similarity distribution chart shown on the overview page. Do you think it would make sense to extract some of the functionality there to a common component or function?

Reply:
I am putting this into a separate issue, together with some cleanup to the other d3 components (as they also have pieces that can be more generic)